<launch>
    <arg name="point_cloud_topic" default="/camera/depth/points"/>
    
    <!-- Global Planner -->
    <node name="global_planner_node" pkg="avoidance" type="global_planner_node" output="screen" 
          args="$(find avoidance)/resource/random_goals"  />
    <!-- <node name="path_handler_node" pkg="avoidance" type="path_handler_node" output="screen" /> -->

    <!-- OctoMap Server -->
    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
        <param name="resolution" value="1.0" />
        <!-- fixed map frame (set to 'map' if SLAM or localization running!) -->
        <param name="frame_id" type="string" value="world" />
        <!-- maximum range to integrate (speedup!) -->
        <param name="sensor_model/max_range" value="9.0" />
        <param name="sensor_model/min" value="0.01" />
        <param name="sensor_model/max" value="0.99" />
        <param name="sensor_model/hit" value="0.9" />
        <param name="sensor_model/miss" value="0.45" />
        <param name="color/r" value="0.1" />
        <param name="color/g" value="0.1" />
        <param name="color/b" value="0.1" />
        <param name="color/a" value="1.0" />
        <!-- Filter out obstacles which are lower than 1 meter -->
        <param name="occupancy_min_z" value="1.0" />
        <param name="height_map" value="false" />
        <param name="publish_free_space" value="false" />
        <!-- data source to integrate (PointCloud2) -->
        <remap from="cloud_in" to="$(arg point_cloud_topic)" />
    </node>
</launch>
